analyses/fusion_filtering/01-fusion-filtering.R

```diff
@@ -0,0 +1,259 @@
+head(sf)
+sf.total <- unique(sf[,c('FusionName','Sample','Caller','Fusion_Type','JunctionReadCount','SpanningFragCount','Confidence')])
+
+#sf.total<-aggregate(list(sf.total$JunctionReadCount,sf.total$JunctionReadCount), list(sf.total$Sample,sf.total$FusionName,sf.total$Caller,sf.total$Fusion_Type, sf.total$Confidence,sf.total$SpanningFragCount) ,sum)
+#colnames(sf.total)<-c('Sample','FusionName','Caller','Fusion_Type','Confidence','SpanningFragCount','JunctionReadCount')
+
+
+
+#IGLC6 is expressed in rnaseq
+sf.total$FusionName<-sub("IGL@","IGLC6",sf.total$FusionName)
+sf.total$FusionName<-sub("IGL-@","IGLC6",sf.total$FusionName)
+
+# arriba fusion
+ar <- read.delim(gzfile(paste0(file.path("../../data/pbta-fusion-arriba.tsv.gz"))),stringsAsFactors=F,,header=T,sep="\t")
+#remove false positives through supporting reads
+ar<-ar[-(which(ar$discordant_mates-(ar$split_reads1+ar$split_reads2) >10 |ar$split_reads1+ar$split_reads2==0)),]
+ar$LeftBreakpoint <- gsub('^chr','',ar$breakpoint1)
+ar$RightBreakpoint <- gsub('^chr','',ar$breakpoint2)
+ar$Fusion_Type<-ar$reading_frame
+ar$Fusion_Type[grep("out-of-frame",ar$Fusion_Type)]<-"frameshift"
+ar$Fusion_Type[-which(ar$Fusion_Type %in% c("in-frame","frameshift"))] <- 'other'
+
+ar$Caller <- 'arriba'
+ar$Sample <-ar$tumor_id
+ar$FusionName <-paste0(gsub(",","/",ar$gene1),"--",gsub(",","/",ar$gene2))
+ar$SpanningFragCount<-ar$discordant_mates
+ar$JunctionReadCount<-ar$split_reads1+ar$split_reads2
+ar$Confidence<-ar$confidence
+ar.total <- unique(ar[,c('FusionName','Sample','Caller','Fusion_Type','JunctionReadCount','SpanningFragCount','Confidence')])
+
+#ar.total<-aggregate(list(ar.total$JunctionReadCount,ar.total$SpanningFragCount), list(ar.total$Sample,ar.total$FusionName,ar.total$Caller,ar.total$Fusion_Type, ar.total$Confidence) ,sum)
+#colnames(ar.total)<-c('Sample','FusionName','Caller','Fusion_Type','Confidence','SpanningFragCount','JunctionReadCount')
+
+
+
+#reassign IGL@ gene name
+ar.total$FusionName<-sub("IGL@","IGLC6",ar.total$FusionName)
+ar.total$FusionName<-sub("IGL-@","IGLC6",ar.total$FusionName)
+
+
+#gather read throughs
+
+sf.rt <- unique(sf[grep('readthrough|neighbors|GTEx_Recurrent|BodyMap|DGD_PARALOGS|HGNC_GENEFAM|ConjoinG',sf$annots,ignore.case=TRUE),])
+#https://arriba.readthedocs.io/en/latest/interpretation-of-results/#frequent-types-of-false-positives
+#not filtered for pcr_fusions
+#only removing read-throughs 
+ar.rt <- unique(ar[grep("read-through",ar$type),])
+ar.rt <- rbind(ar.rt,unique(ar[grep("readthrough|Normal|neighbors|GTEx_Recurrent|BodyMap|DGD_PARALOGS|HGNC_GENEFAM|ConjoinG",ar[,27],ignore.case=TRUE),]))
+rts <- unique(c(sf.rt$FusionName, ar.rt$FusionName))
+rts.rev <- unique(unlist(lapply(strsplit(rts, '--'), FUN = function(x) paste0(x[2],'--',x[1]))))
+rts <- unique(c(rts, rts.rev))
+
+
+#merge callers
+all.callers<-rbind(ar.total, sf.total)
+
+#histology
+clin<-read.delim(paste0(file.path("../../data/pbta-histologies.tsv")),stringsAsFactors = F,sep="\t")
```

Comment:
Unfortunately I'm not going to be able to go through this whole script today, but this came to my attention because of https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/94 and wanted to discuss before making changes to the README.

I think this change would be equivalent to what you currently have:

```suggestion
clin<-read.delim("../../data/pbta-histologies.tsv",stringsAsFactors = F,sep="\t")
```
or if you wanted to use `file.path`:

```suggestion
clin<-read.delim(file.path("..", "..", "data", "pbta-histologies.tsv"),stringsAsFactors = F,sep="\t")
```

As far as I know, you don't need to treat the symlinks as any kind of special case when reading them into R. Provided `bash download-data.sh` ran to completion, you can use the path to the symlinks in `../../data` and that will read in the equivalent file from the most recent release.

Reply:
Please let me know if that works and if my explanation was helpful, it will help me formulate changes for #94.

Reply:
I updated the file.path to file.path("..", "..", "data","pbta-fusion-starfusion.tsv.gz") still doesnt read the file and fails CI , am I missing something?  
I had the paste0 from the time I was running paste0("../../data/",Sys.readlink2("../../data/pbta-fusion-starfusion.tsv.gz")) to get the symbolic link for gzfile().

Reply:
Ah, okay looking at the CI details: https://circleci.com/gh/AlexsLemonade/OpenPBTA-analysis/252?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

I think I might know what's going on now that I have that context. When calling this in CI with:
```sh
./scripts/run_in_ci.sh Rscript analyses/fusion_filtering/01-fusion-filtering.R
```

The working directory will be the root directory of the repository and you will not have to traverse upward to `../../data`. It would be `data/pbta-fusion-starfusion.tsv.gz` rather than `../../data/pbta-fusion-starfusion.tsv.gz`.

I went back to another PR where the Rscript was being run similarly: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/54/files#diff-1d37e48f9ceff6d8030570cd36286a61R20
and `data` was used: https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/54/files#diff-ad4abc42962164e2453a997a53b3505fR49

Reply:
oh yes that makes sense, thanks!